src/pages/home/library/SongList.tsx

```diff
@@ -64,16 +64,35 @@ export default function SongList({ totalCountOfSongs, query }: SongListProps) {
       remainingSongsOnLastPage > 0 ? remainingSongsOnLastPage : rowsPerPage;
   }
 
+  const [streamUrlMissing, setStreamUrlMissing] = useState(true);
+
   const {
     data: songData = [],
     isLoading,
     isSuccess,
-  } = useGetSongsQuery({
-    ownerIds: ["me"],
-    offset: page - 1,
-    limit: songsToRequest,
-    phrase: query,
-  });
+  } = useGetSongsQuery(
+    {
+      ownerIds: ["me"],
+      offset: page - 1,
+      limit: songsToRequest,
+      phrase: query,
+    },
+    {
+      // Refetch songs every minute if the streamUrl is missing for any song
+      pollingInterval: streamUrlMissing ? 60000 : undefined,
+    }
+  );
+
+  // Checks if any of the songs are missing a streamUrl
+  useEffect(() => {
+    const someSongsNotTranscoded = songData.some((song) => !song.streamUrl);
+
+    if (someSongsNotTranscoded) {
+      setStreamUrlMissing(true);
+    } else {
+      setStreamUrlMissing(false);
+    }
+  }, [songData, streamUrlMissing]);
```

Comment:
This can be simplified a bit like this:
```
useEffect(() => {
  const someSongsNotTranscoded = songData.some((song) => !song.streamUrl);
  setStreamUrlMissing(someSongsNotTranscoded);
}, [songData]);
```
or skip the variable and do it all at once? 🤷🏽‍♂️


Also, i think you can remove `streamUrlMissing` from the dependency array because the only place this is being changed is in this useEffect so i dont think you need it